public/js/NeighbourPlot.js

```diff
@@ -0,0 +1,199 @@
+var d3 = require('d3');
+require('d3-tip')(d3);
+
+/**
+ * Draws PCA scatterplot of the data Clicking on scatterplot points will
+ * trigger updates of the neighbour image gallery and corresponding sample
+ * lineplot.
+ *
+ *
+ * @constructor
+ * @param {array} sampleData - The sample data for the screen. Each element
+ * in the array is an instance of a Sample document.
+ * @param {string} element - The ID of the target div for this plot.
+ * @param {LinePlot} lineplot - LinePlot that will update when scatterplot
+ * points are clicked.
+ * @param {NeighbourImages} neighbourImages - NeighbourImages object that will
+ * update when scatterplot points are clicked.
+ */
+function NeighbourPlot(sampleData, element, lineplot, neighbourImages) {
+    this.sampleData = sampleData;
+    this.lineplot = lineplot;
+    this.neighbourImages = neighbourImages;
+
+    this.element = element;
+    this.fullWidth = 700;
+    this.fullHeight = 500;
+    this.transitionDuration = 125;
+    this.inactivePointRadius = 5;
+    this.activePointRadius = 7;
+    this.xAxisTicks = 10;
+    this.yAxisTicks = 10;
+
+    this.margin = {top: 10, right: 40, bottom: 30, left: 40};
+    this.width = this.fullWidth - this.margin.left - this.margin.right;
+    this.height = this.fullHeight - this.margin.top - this.margin.bottom;
+
+    this.drawScatterplot();
+}
+
+/**
+ * This function draws the PCA plot.
```

Comment:
function -> method? This would keep it consistent with common OO terminology and help my brain to not hurt. =P